src/iroh.rs

```diff
@@ -474,159 +422,281 @@ impl VeilidIrohBlobs {
-        let map = self.collection_hashes.read().await;
-        if let Some(hash) = map.get(collection_name) {
-            Ok(*hash)
-        } else {
-            Err(anyhow!("Collection {} not found", collection_name))
+        // Retrieve the tag from the store instead of using the in-memory cache
+        match self.get_tag(collection_name).await {
+            Ok(hash) => Ok(hash),
+            Err(e) => {
+                println!("Error retrieving tag for collection: {:?}", e);
+                Err(anyhow!("Collection {} not found", collection_name))
+            }
+        }
+    }
+    pub async fn store_tag(&self, collection_name: &str, collection_hash: &Hash) -> Result<()> {
+        println!("Storing tag for collection: {} with hash: {:?}", collection_name, collection_hash);
+        
+        // Store the tag
+        self.store.set_tag(
+            collection_name.to_string().into(), 
+            Some(HashAndFormat::new(*collection_hash, BlobFormat::Raw))
+        ).await?;
+    
+        // Introducing a short delay to ensure that tag is persisted
+        tokio::time::sleep(Duration::from_millis(100)).await;
+    
+        // Verify that the tag was actually stored
+        match self.get_tag(collection_name).await {
+            Ok(stored_hash) => {
+                if stored_hash == *collection_hash {
+                    println!("Tag successfully stored for collection: {}", collection_name);
+                } else {
+                    return Err(anyhow!("Mismatch in stored tag for collection: {}", collection_name));
+                }
+            },
+            Err(e) => {
+                return Err(anyhow!("Failed to retrieve tag after storing for collection: {}. Error: {:?}", collection_name, e));
+            }
+        }
+        
+        Ok(())
+    }
+    
+    
+    
+    
+    pub async fn get_tag(&self, collection_name: &str) -> Result<Hash> {
+        println!("Retrieving tag for collection: {}", collection_name);
+        let mut tags = self.store.tags().await?;
+    
+           
+        while let Some(tag_result) = tags.next() {
+            match tag_result {
+                Ok((tag, hash_and_format)) => {
+                    // Logging for better insight
+                    println!("Checking tag: {:?}", tag);
+
+                    // Extract the bytes and convert them to a string
+                    let tag_bytes = tag.borrow();
+                    let tag_str = std::str::from_utf8(tag_bytes).unwrap_or("");
+
+                    // Compare the extracted tag string with collection_name
+                    if tag_str == collection_name {
+                        println!("Found matching tag for collection: {}", collection_name);
+                        return Ok(hash_and_format.hash);
+                    } else {
+                        println!("Tag {} does not match {}", tag_str, collection_name);
+                    }
+                }
+                Err(e) => {
```

Comment:
Prefer to use `tag_result?` or `.map_err` to these match statements

Reply:
Switched to using the ? operator and map_err
ff27d38067a00229ac9fb03d159bc4ade2b228aa